README.md

```diff
@@ -116,25 +165,75 @@ j2 = Rotations.jacobian(q, p) # How does the rotated point q*p change w.r.t. the
 
-8. **Euler Angles - Three-axis rotations** `RotXYZ{T}`, `RotXYX{T}`, etc
+10. **Euler Angles - Three-axis rotations** `RotXYZ{T}`, `RotXYX{T}`, etc
 
     A composition of 3 cardinal axis rotations is typically known as a Euler
     angle parameterization of a 3D rotation. The rotations with 3 unique axes,
     such as `RotXYZ`, are said to follow the [**Tait Bryan**](https://en.wikipedia.org/wiki/Euler_angles#Tait.E2.80.93Bryan_angles) angle ordering,
     while those which repeat (e.g. `EulerXYX`) are said to use [**Proper Euler**](https://en.wikipedia.org/wiki/Euler_angles#Conventions) angle ordering.
 
     Like the two-angle versions, the order of application to a vector is right-to-left, so that `RotXYZ(x, y, z) * v == RotX(x) * (RotY(y) * (RotZ(z) * v))`.  This may be interpreted as an "extrinsic" rotation about the Z, Y, and X axes or as an "intrinsic" rotation about the X, Y, and Z axes.  Similarly, `RotZYX(z, y, x)` may be interpreted as an "extrinsic" rotation about the X, Y, and Z axes or an "intrinsic" rotation about the Z, Y, and X axes. 
+    
+### The Rotation Error state and Linearization
+It is often convenient to consider perturbations or errors about a particular 3D rotation, such as applications in state estimation or optimization-based control. Intuitively, we expect these errors to live in three-dimensional space. For globally non-singular parameterizations such as unit quaternions, we need a way to map between the four parameters of the quaternion to this three-dimensional plane tangent to the four-dimensional hypersphere on which quaternions live. 
+
+There are several of these maps provided by Rotations.jl:
+* `ExponentialMap`: A very common mapping that uses the quaternion
+exponential and the quaternion logarithm. The quaternion logarithm
+converts a 3D rotation vector (i.e. axis-angle vector) to a unit quaternion.
+It tends to be the most computationally expensive mapping.
+
+* `CayleyMap`: Represents the differential quaternion using Rodrigues
+parameters. This parameterization goes singular at 180° but does not
+inherit the sign ambiguities of the unit quaternion. It offers an
+excellent combination of cheap computation and good behavior.
+
+* `MRPMap`: Uses Modified Rodrigues Parameters (MRPs) to represent the
+differential unit quaternion. This mapping goes singular at 360°.
+
+* `QuatVecMap`: Uses the vector part of the unit quaternion as the
+differential unit quaternion. This mapping also goes singular at 180° but is
+the computationally cheapest map and often performs well.
+
+Rotations.jl provides the `RotationError` type for representing rotation errors, that act just like a `SVector{3}` but carry the nonlinear map used to compute the error, which can also be used to convert the error back to a `UnitQuaternion` (and, by extention, any other 3D rotation parameterization). The following methods are useful for computing `RotationError`s and adding them back to the nominal rotation:
```

Comment:
Very nice. I feel like this is really a much clearer design than having extra type parameters into `UnitQuaternion`.